nucypher/blockchain/eth/token.py

```diff
@@ -159,6 +161,24 @@ class Stake:
     class StakingError(Exception):
         """Raised when a staking operation cannot be executed due to failure."""
 
+    class Status(Enum):
+        """
+        Sub-stake status.
+        """
+        INACTIVE = 1   # Unlocked inactive
+        UNLOCKED = 2   # Unlocked active
+        LOCKED = 3     # Locked but not editable
+        EDITABLE = 4   # Editable
+        DIVISIBLE = 5  # Editable and divisible
+
+        def is_child(self, other: 'Status') -> bool:
+            if other == self.INACTIVE:
+                return self == self.INACTIVE
+            elif other == self.UNLOCKED:
+                return self.value <= self.UNLOCKED.value
+            else:
+                return self.value >= other.value
```

Comment:
I find this logic a bit hard to follow and weird for an `Enum`. You can simplify it by using an `IntEnum` rather than an `Enum` which confers `int`-like operations on the class itself.

Then it can be simplified to (I think):
```python
return self >= other or self < (self.UNLOCKED and self)
```

Or if you don't like condensed logic:
```python
if not other == self.UNLOCKED:
    return self >= other
return self < self.UNLOCKED
```

Reply:
I'm not sure, in that case `self.INACTIVE == 1` will be `True` which looks weird for that `Status`

Reply:
In that case, you do `a_status.name`.

Reply:
I understand, I mean `Status` instance will behave not as some class but as `int` which is not great for me